docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1021 @@
+endif::teardown_method[]
+
+=== Add a Gemfile
+Gemfiles allow Ruby's Bundler to maintain the dependencies for your plugin.
+Currently, all we'll need is the Logstash gem, for testing, but if you require
+other gems, you should add them in here.
+
+TIP: See http://bundler.io/gemfile.html[Bundler's Gemfile page] for more details.
+
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+source 'https://rubygems.org'
+gemspec
+gem "logstash", :github => "elasticsearch/logstash", :branch => "{branch}"
+----------------------------------
+
+=== Add a `gemspec` file
+Gemspecs define the Ruby gem which will be built and contain your plugin.
+
+TIP: More information can be found on the
+http://guides.rubygems.org/specification-reference/[Rubygems Specification page].
+
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+Gem::Specification.new do |s|
+  s.name = 'logstash-{plugintype}-{pluginname}'
+  s.version = '0.1.0'
+  s.licenses = ['Apache License (2.0)']
+  s.summary = "This {plugintype} does x, y, z in Logstash"
+  s.description = "This gem is a logstash plugin required to be installed on top of the Logstash core pipeline using $LS_HOME/bin/plugin install gemname. This gem is not a stand-alone program"
+  s.authors = ["Elasticsearch"]
+  s.email = 'info@elasticsearch.com'
+  s.homepage = "http://www.elasticsearch.org/guide/en/logstash/current/index.html"
+  s.require_paths = ["lib"]
+
+  # Files
+  s.files = `git ls-files`.split($\)
+   # Tests
+  s.test_files = s.files.grep(%r{^(test|spec|features)/})
+
+  # Special flag to let us know this is actually a logstash plugin
+  s.metadata = { "logstash_plugin" => "true", "logstash_group" => "{plugintype}" }
+
+  # Gem dependencies
+  s.add_runtime_dependency 'logstash', '>= 1.4.0', '< 2.0.0'
+  s.add_development_dependency 'logstash-devutils'
+end
+----------------------------------
+
+==== Runtime & Development Dependencies
+At the bottom of the `gemspec` file is a section with a comment:
+`Gem dependencies`.  This is where any other needed gems must be mentioned. If
+a gem is necessary for your plugin to function, it is a runtime dependency. If
+a gem are only used for testing, then it would be a development dependency.
+
+[NOTE]
+=========================
+You can also have versioning requirements for your dependencies, e.g.
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  # Gem dependencies
+  s.add_runtime_dependency 'logstash', '>= 1.4.0', '< 2.0.0'
+  s.add_development_dependency 'logstash-devutils'
+----------------------------------
+…has a runtime dependency on the core Logstash gem and requires that it have a
+version number greater than or equal to version 1.4.0 `'>= 1.4.0'`, and less
+than version 2.0 `'< 2.0.0'`.
+=========================
+
+
+IMPORTANT: All plugins have a runtime dependency on the `logstash` core gem, and a
+development dependency on `logstash-devutils`.
+
+=== Add Tests
```

Comment:
This seems like part of the overall process, the following section (at the same level) is what you need to do once you've done the preceding steps.